src/states/elder/mod.rs

```diff
@@ -1591,12 +1608,33 @@ impl Approved for Elder {
             self.finalise_prefix_change()?;
             self.send_event(Event::SectionMerged(*elders_info.prefix()), outbox);
         } else if self_sec_update {
+            if !elders_info.members().contains(&self.full_id.public_id()) {
+                return Ok(Transition::Demote);
+            }
             self.reset_parsec()?;
         }
 
-        self.update_neighbour_connections(neighbour_change, outbox);
+        for pub_id in &elders_change.own_added {
+            self.send_event(Event::NodeAdded(*pub_id.name()), outbox);
+        }
+
+        for pub_id in &elders_change.own_removed {
+            self.send_event(Event::NodeLost(*pub_id.name()), outbox);
+        }
+
+        self.update_neighbour_connections(elders_change, outbox);
 
         if self_sec_update {
+            for pub_id in self.chain.adults_and_infants() {
```

Comment:
There is a problem here:
Only one side of the split will do that, because the other side will have the same SectionInfo coming out, and so self_sec_update will be false. I've meant to refactor that code for a while (and did some of it so we separate our own handling of SectionInfo and neighbour_infos.